src/submission/validation-clinical/therapy.ts

```diff
@@ -17,42 +17,234 @@
 
 export const validate = async (
   therapyRecord: DeepReadonly<SubmittedClinicalRecord>,
   existentDonor: DeepReadonly<Donor>,
   mergedDonor: Donor,
+  submittedRecords: DeepReadonly<ClinicalSubmissionRecordsByDonorIdMap>,
 ): Promise<SubmissionValidationOutput> => {
   // ***Basic pre-check (to prevent execution if missing required variables)***
   if (!therapyRecord || !mergedDonor || !existentDonor) {
     throw new Error("Can't call this function without a registerd donor & therapy record");
   }
 
-  const errors: SubmissionValidationError[] = [];
+  let errors: SubmissionValidationError[] = [];
   const treatment = getTreatment(therapyRecord, mergedDonor, errors);
   if (!treatment) return { errors };
-  checkTreatementHasCorrectTypeForTherapy(therapyRecord, treatment, errors);
+  // TODO: Refactor use of push
+  checkTreatmentHasCorrectTypeForTherapy(therapyRecord, treatment, errors);
+
+  const crossFileErrors = await crossFileValidator(
+    mergedDonor,
+    treatment,
+    therapyRecord,
+    submittedRecords,
+  );
+
+  errors = [...errors, ...crossFileErrors];
   return { errors };
 };
 
-export function checkTreatementHasCorrectTypeForTherapy(
+const crossFileValidator = async (
+  donor: Donor,
+  treatment: DeepReadonly<Treatment>,
+  therapyRecord: DeepReadonly<SubmittedClinicalRecord>,
+  // submitted records needed to validate current submission
+  submittedRecords: DeepReadonly<ClinicalSubmissionRecordsByDonorIdMap>,
+) => {
+  const isRadiationRecord = Object.values(RadiationFieldsEnum).some(field =>
+    Object.keys(therapyRecord).includes(field),
+  );
+
+  const radiationErrors: SubmissionValidationError[] = isRadiationRecord
+    ? await validateRadiationRecords(donor, treatment, therapyRecord, submittedRecords)
+    : [];
+
+  const crossFileErrors = [...radiationErrors];
+
+  return crossFileErrors;
+};
+
+interface DonorResult {
+  donors: DeepReadonly<Donor>[];
+}
+
+const emptyDonorResult: DonorResult = { donors: [] };
+
+const validateRadiationRecords = async (
+  donor: Donor,
+  treatment: DeepReadonly<Treatment>,
+  therapyRecord: DeepReadonly<SubmittedClinicalRecord>,
+  submittedRecords: DeepReadonly<ClinicalSubmissionRecordsByDonorIdMap>,
+) => {
+  const { programId } = donor;
+
+  const {
+    clinicalInfo: {
+      submitter_donor_id: relatedTreatmentDonorId,
+      submitter_treatment_id,
+      treatment_type,
+    },
+  } = treatment;
+
+  const {
+    submitter_donor_id: submittedTherapyDonorId,
+    radiation_boost,
+    reference_radiation_treatment_id,
+  } = therapyRecord;
+
+  // Compare across other Submissions
+  let submittedTreatments: DeepReadonly<ClinicalInfo>[] = [];
+
+  for (const donorId in submittedRecords) {
+    const submittedTreatmentRecords = [...submittedRecords[donorId].treatment] as ClinicalInfo[];
+    if (submittedTreatmentRecords && submittedTreatmentRecords.length)
+      submittedTreatments = [...submittedTreatments, ...submittedTreatmentRecords];
+  }
```

Comment:
``` typescript

  for (const donorId in submittedRecords) {
    const submittedTreatmentRecords = submittedRecords[donorId].treatment
    if (submittedTreatmentRecords?.length)
      submittedTreatments = [...submittedTreatments, ...submittedTreatmentRecords];
  }
  ```
  
  I think this works the same (ignore typings for now).
  `submittedRecord[donorId].treatment` looks like it should be an array already, so the spread new array thing is redundant

Reply:
I will give a second look, given I used 'as' this is definitely a crutch specifically to manage Types

Reply:
i don't think the `as` assertion is needed , submittedRecords is typed already so submittedTreatmentRecords should be grand as is

Reply:
Yep looks like this works